modules/src/core/ics04_channel/handler/timeout.rs

```diff
@@ -59,25 +58,18 @@ pub fn process<HostFunctions: HostFunctionsProvider>(
 
     // check that timeout height or timeout timestamp has passed on the other end
     let proof_height = msg.proofs.height();
-    let packet_height = packet.timeout_height;
-
-    if (!packet.timeout_height.is_zero()) && packet_height > proof_height {
-        return Err(Error::packet_timeout_height_not_reached(
-            packet.timeout_height,
-            proof_height,
-        ));
-    }
 
     let consensus_state = ctx
         .consensus_state(&client_id, proof_height)
         .map_err(|_| Error::error_invalid_consensus_state())?;
 
     let proof_timestamp = consensus_state.timestamp();
 
-    let packet_timestamp = packet.timeout_timestamp;
-    if let Expiry::Expired = packet_timestamp.check_expiry(&proof_timestamp) {
-        return Err(Error::packet_timeout_timestamp_not_reached(
-            packet_timestamp,
+    if !packet.timed_out(&proof_timestamp, proof_height) {
```

Comment:
shouldn't we check that the timestamp isn't zero before doing this?

Reply:
Zero timestamp is represented as `None` internally and 
`packet.timed_out`  function checks that with this `self.timeout_timestamp != Timestamp::none()`